src/services/pathContents/branch/dir/usePrefetchBranchDirEntry.spec.tsx

```diff
@@ -0,0 +1,371 @@
+            results: [
+              {
+                __typename: 'PathContentDir',
+                name: 'src',
+                path: null,
+                percentCovered: 0.0,
+                hits: 4,
+                misses: 2,
+                lines: 7,
+                partials: 1,
+              },
+            ],
+          },
+        },
+      },
+    },
+  },
+}
+
+const mockDataUnknownPath = {
+  owner: {
+    username: 'codecov',
+    repository: {
+      __typename: 'Repository',
+      repositoryConfig: {
+        indicationRange: {
+          upperRange: 80,
+          lowerRange: 60,
+        },
+      },
+      branch: {
+        head: {
+          pathContents: {
+            message: 'path cannot be found',
+            __typename: 'UnknownPath',
+          },
+        },
+      },
+    },
+  },
+}
+
+const mockDataMissingCoverage = {
+  owner: {
+    username: 'codecov',
+    repository: {
+      __typename: 'Repository',
+      repositoryConfig: {
+        indicationRange: {
+          upperRange: 80,
+          lowerRange: 60,
+        },
+      },
+      branch: {
+        head: {
+          pathContents: {
+            message: 'files missing coverage',
+            __typename: 'MissingCoverage',
+          },
+        },
+      },
+    },
+  },
+}
+
+const mockDataRepositoryNotFound = {
+  owner: {
+    repository: {
+      __typename: 'NotFoundError',
+      message: 'repository not found',
+    },
+  },
+}
+
+const mockDataOwnerNotActivated = {
+  owner: {
+    repository: {
+      __typename: 'OwnerNotActivatedError',
+      message: 'owner not activated',
+    },
+  },
+}
+
+const mockUnsuccessfulParseError = {}
+
+const queryClient = new QueryClient({
+  defaultOptions: { queries: { retry: false } },
+})
+
+const wrapper: React.FC<React.PropsWithChildren> = ({ children }) => (
+  <QueryClientProvider client={queryClient}>
+    <MemoryRouter initialEntries={['/gh/codecov/test-repo/tree/main/src']}>
+      <Route path="/:provider/:owner/:repo/tree/:branch/:path+">
+        {children}
+      </Route>
+    </MemoryRouter>
+  </QueryClientProvider>
+)
+
+const server = setupServer()
+beforeAll(() => {
+  server.listen()
+})
+
+beforeEach(() => {
+  server.resetHandlers()
+  queryClient.clear()
+})
+
+afterAll(() => {
+  server.close()
+})
+
+interface SetupArgs {
+  isMissingCoverage?: boolean
+  isUnknownPath?: boolean
+  isRepositoryNotFoundError?: boolean
+  isOwnerNotActivatedError?: boolean
+  isUnsuccessfulParseError?: boolean
+}
+
+describe('usePrefetchBranchDirEntry', () => {
+  function setup({
+    isMissingCoverage = false,
+    isUnknownPath = false,
+    isRepositoryNotFoundError = false,
+    isOwnerNotActivatedError = false,
+    isUnsuccessfulParseError = false,
+  }: SetupArgs) {
+    server.use(
+      graphql.query('BranchContents', (req, res, ctx) => {
+        if (isMissingCoverage) {
+          return res(ctx.status(200), ctx.data(mockDataMissingCoverage))
+        } else if (isUnknownPath) {
```

Comment:
nit: don't need else if when returning in if blocks

Reply:
This is interesting lol. Wouldn't the calculation be `\n` versus ` else ` making the former smaller?

Reply:
Nope. You are right! I tested it:
```
$ ls -la | grep DirEntry.spec.tsx
-rw-r--r--  1 spencer  staff  9218  6 May 16:01 usePrefetchBranchDirEntry.spec.tsx
$ ls -la | grep DirEntry.spec.tsx
-rw-r--r--  1 spencer  staff  9202  6 May 16:02 usePrefetchBranchDirEntry.spec.tsx
```

top is newline bottom is else if